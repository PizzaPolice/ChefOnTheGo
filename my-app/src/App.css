/*
--- Basic Styling
*/

* {
  box-sizing: border-box;
}

html,
body {
    font-family: 'Montserrat', sans-serif;
    height: 100%;
    padding: 10px;
    margin: 0;
    background-color: #d3d3d3 /*light grey*/

}

/*
  Outer panel
*/
.outer-panel {
    width: 940px;
    height: 690px;
    margin: 50px auto; /*very top margin*/
    background: #FFFFFF;
    box-shadow: 1px 2px 3px 0px rgba(0,0,0,0.10);
    border-radius: 6px;
    display: flex;
    flex-direction: column
}

/* Panel Body */
.panel-body {
  padding: 45px 80px 0;
  flex: 1;
}

.title {
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 40px;
  color: #2e2e2e;
}

.blue-border {
  border: 1px solid rgb(110, 178, 251);
}

.input-line{
  margin-top: 10px;
}

input[type='radio'] {
  display: block;
}

.radio-input {
  margin-top: 20px;
}


/* Input Fields */
.input-fields {
  display: flex;
  justify-content: space-between;
}

.input-fields label {
  display: block;
  margin-top: 10px;
  margin-bottom: 7px; /* margin under "cardholders name" etc*/
  color: #b4b4b4;
}

div[class*='column'] {
  width: 382px;
}

input[type='text'],
input[type='password'] {
  font-size: 16px;
  width: 100%;
  height: 50px;
  padding-right: 40px;
  padding-left: 16px;
  color: rgba(46, 46, 46, .8);
  border: 1px solid rgb(225, 225, 225);
  border-radius: 4px;
  outline: none;
}

input[type='text']:focus,
input[type='password']:focus {
  border-color: rgb(119, 219, 119);
}

/* Panel Footer */
.panel-footer {
    display: flex;
    margin-top: 100px; /* space above bottom panel */
    margin-bottom: 45px;
    /*margin-right: 50px;*/
    width: 100%;
    height: 80px;  
    padding: 0 70px;
    /*background: #FFFFFF;*/
    /*background-color: rgba(239, 239, 239, 0.664);*/
    justify-content: space-between;
    align-items: center;

}

/* Buttons */
.btn {
  font-size: 16px;
  width: 163px;
  height: 48px;
  /*margin-right: 500px;
  margin-left: -50px;  /*move back button*/
  cursor: pointer;
  transition: all .2s ease-in-out;
  letter-spacing: 1px;
  border: none;
  border-radius: 23px;
}

.back-btn {
  color: #f62f5e;
  background: rgb(252, 250, 250);

}

.next-btn {
  color: #fff;
  background: #f62f5e;
}

.btn:focus {
  outline: none;
}

.btn:hover {
  transform: scale(1.1);
}

.shopping-cart {
  width: 800px;
  height: 766px;
  margin: 50px auto; /*very top margin*/
  background: #FFFFFF;
  box-shadow: 1px 2px 3px 0px rgba(0,0,0,0.10);
  border-radius: 6px;

  display: flex;
  flex-direction: column;
}

.title {
  height: 60px;
  border-bottom: 1px solid #E1E8EE;
  padding: 20px 30px;
  color: #5E6977;
  font-size: 18px;
  font-weight: 400;
}

.item {
  padding: 10px 30px;

  display: flex;
}

.item:nth-child(1) {
  /*border-top:  1px solid #E1E8EE;*/
  border-bottom:  1px solid #E1E8EE;
  margin-bottom: -10px;
  margin-top: -40px;
}

.item:nth-child(2) {
  /*border-top:  1px solid #E1E8EE;*/
  border-bottom:  1px solid #E1E8EE;
  margin-top: 10px;
}

/*food 3*/
.item:nth-child(3) {
  /*border-top:  1px solid #E1E8EE;*/
  border-bottom:  1px solid #E1E8EE;
  /*just to make buttons look decent, won't work if another food item is added */
  margin-bottom: 240px; 
  margin-top: 10px;
}


/* Buttons -  Delete and Like */
.buttons {
  position: relative;
  padding-top: 30px;
  margin-right: 60px;
}

.is-active {
  animation-name: animate;
  animation-duration: .8s;
  animation-iteration-count: 1;
  animation-timing-function: steps(28);
  animation-fill-mode: forwards;
}

@keyframes animate {
  0%   { background-position: left;  }
  50%  { background-position: right; }
  100% { background-position: right; }
}


/* Product Description */
.description {
  padding-top: 5px;
  margin-right: 60px;
  width: 115px;
}

.description span {
  display: flex;
  font-size: 14px;
  color: #43484D;
  font-weight: 400;
}

.description span:first-child {
  margin-bottom: 5px;
}
.description span:last-child {
  font-weight: 300;
  margin-top: 8px;
  color: #86939E;
}

/* Product Quantity */
.quantity {
  padding-top: 12px;
  padding-left: 120px;
  margin-right: 90px;
  margin-left: 150px;
  margin-bottom: 50px;
}
.quantity input {
  border: none;
  text-align: center;
  width: 32px;
  font-size: 16px;
  color: #43484D;
  font-weight: 300;
}

.minus-btn img {
  margin-bottom: 3px;
}
.plus-btn img {
  margin-top: 2px;
}
button:focus,
input:focus {
  outline:0;
}

/* Total Price */
.total-price {
  width: 83px;
  margin-left: 160px;
  padding-top: 27px;
  text-align: right;
  font-size: 16px;
  color: #43484D;
  font-weight: 100;
}

/* Responsive */
@media (max-width: 800px) {
  .shopping-cart {
    width: 100%;
    height: auto;
    overflow: hidden;
  }
  .item {
    height: auto;
    flex-wrap: wrap;
    justify-content: center;
  }

  .quantity,
  .description {
    width: 100%;
    text-align: center;
    margin: 6px 0;
  }
  .buttons {
    margin-right: 20px;
  }
}


.payment-method {
  display: flex;
  margin-bottom: 40px; /*margin below visa/paypal*/
  justify-content: space-between;
}

.method {
  display: flex;
  flex-direction: column;
  width: 382px;
  height: 122px;
  padding-top: 20px;
  cursor: pointer;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: rgb(249, 249, 249);
  justify-content: center;
  align-items: center;
}

.blue-border {
  border: 1px solid rgb(110, 178, 251);
}

.card-logos {
  display: flex;
  width: 150px;
  justify-content: space-between;
  align-items: center;
}

.radio-input {
  margin-top: 20px;
}

input[type='radio'] {
  display: inline-block;
}


/* Input Fields */
.input-fields {
  display: flex;
  justify-content: space-between;
}

.input-fields label {
  display: block;
  margin-top: 10px;
  margin-bottom: 7px; /* margin under "cardholders name" etc*/
  color: #b4b4b4;
}

.warning {
  border-color: #f62f5e !important;
}

.info {
  font-size: 12px;
  font-weight: 300;
  display: block;
  margin-top: 50px;
  opacity: .5;
  color: #2e2e2e;
}

div[class*='column'] {
  width: 382px;
}

/*food order quantity*/
input[type='text'],
input[type='password'] {

  font-size: 16px;
  width: 100%;
  height: 30px;
  padding-right: 20px;
  padding-left: 16px;
  color: rgba(46, 46, 46, .8);
  border: 1px solid rgb(225, 225, 225);
  border-radius: 4px;
  outline: none;
  
}

input[type='text']:focus,
input[type='password']:focus {
  border-color: rgb(119, 219, 119);
}
