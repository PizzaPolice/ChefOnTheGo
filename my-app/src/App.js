import React, { Component } from 'react';
import { BrowserRouter, Route} from "react-router-dom";
import {Provider} from 'react-redux'

import Login from './components/login.js';
import Signup from './components/signup.js'
import Customize from './components/customize.js';
import Chat from './components/chat.jsx';
import ListChefs from './components/list_chefs.js';
import Review from './components/review.jsx';
import EditDishes from './components/edit_dishes.jsx';
import InactiveChef from './components/inactive_chef.jsx';
import ActiveChef from './components/active_chef.jsx';
import FoodOrder from './components/food_order.jsx';
import Checkout from './components/checkout.jsx';
import OrderStatus from './components/order_status.jsx';

require('./App.css');

//added to create store
const store = createStore(rootReducers);
console.log(store.getState());
store.subscribe(() => console.log('store',store.getState()))
store.dispatch(addCharacterById(3));


class App extends Component {  
  render(){
    return (

      //added 
    <Provider store={store}> 

      <BrowserRouter>
      <div>
          <Route exact path="/" component={Login}/>
          <Route path="/login" component={Login}/>
          <Route path="/signup" component={Signup}/>
          <Route path="/customize" component={Customize}/>
          <Route path="/list_chefs" component={ListChefs}/>
          <Route path="/review" component={Review}/>
          <Route path="/inactive_chef" component={InactiveChef}/>
          <Route path="/edit_dishes" component={EditDishes}/>
          <Route path="/chat" component={Chat}/>
          <Route path="/active_chef" component={ActiveChef}/>
          <Route path="/food_order" component={FoodOrder}/>
          <Route path="/checkout" component={Checkout}/>
          <Route path="/order_status" component={OrderStatus}/>
      </div>
      </BrowserRouter>
 
 </Provider>
  , document.getElementById('root')

    );
  }
}

export default App;
