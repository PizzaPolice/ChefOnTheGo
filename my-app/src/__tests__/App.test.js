import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import App from '../App.js';

/*
it('App renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
*/

describe('App renders without crashing', () => {
  it('App should render without throwing an error', () => {
      expect(shallow(<App />).exists(<form className='App'></form>)).toBe(true)
    })
  });

