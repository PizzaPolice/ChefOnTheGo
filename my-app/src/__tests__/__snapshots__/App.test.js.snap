// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing should render initial layout 1`] = `
Array [
  <Unknown>
    <div>
        <ul>
            <li>
                <Link
                    replace={false}
                    to="/chat"
                >
                    Chat
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/inactive_chef"
                >
                    Inactive Chef
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/edit_dishes"
                >
                    Edit Dishes
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/food_order"
                >
                    Food Order
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/active_chef"
                >
                    Active Chef
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/checkout"
                >
                    Checkout
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/order_status"
                >
                    Order Status
                </Link>
            </li>
        </ul>
        <hr />
        <Route
            component={[Function]}
            exact={true}
            path="/"
        />
        <Route
            component={[Function]}
            path="/login"
        />
        <Route
            component={[Function]}
            path="/signup"
        />
        <Route
            component={[Function]}
            path="/customize"
        />
        <Route
            component={[Function]}
            path="/list_chefs"
        />
        <Route
            component={[Function]}
            path="/review"
        />
        <Route
            component={[Function]}
            path="/inactive_chef"
        />
        <Route
            component={[Function]}
            path="/edit_dishes"
        />
        <Route
            component={[Function]}
            path="/chat"
        />
        <Route
            component={[Function]}
            path="/active_chef"
        />
        <Route
            component={[Function]}
            path="/food_order"
        />
        <Route
            component={[Function]}
            path="/checkout"
        />
        <Route
            component={[Function]}
            path="/order_status"
        />
    </div>
</Unknown>,
]
`;
